pkg/receive/handler.go

```diff
@@ -142,6 +148,38 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 				Help: "The number of times to replicate incoming write requests.",
 			},
 		),
+		writeTimeseriesTotal: promauto.With(o.Registry).NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_timeseries_total",
+				Help:      "The number of timeseries received in the incoming write requests.",
+			}, []string{"code", "method", "tenant"},
+		),
+		writeSamplesTotal: promauto.With(o.Registry).NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_samples_total",
+				Help:      "The number of sampled received in the incoming write requests.",
+			}, []string{"code", "method", "tenant"},
+		),
```

Comment:
I would like to propose to transform these two metrics into histograms or summaries. What are the recommendations? 

In case histogram is a good choice, I think it would be good to let the user configure the buckets and we provide some sensible defaults. Maybe a `prometheus.ExponentialBucket` would be a good starter?

Reply:
Histograms with hardcoded (initally) sane buckets could do the work.  👍🏽 

I would create manually exponential buckets - to manually check what we might find useful. Also probably we could define (if not yet) some sane limits for shape of request which can configure max bucket (: